### PR TITLE
Add 'what's new' docs for 3.5

### PR DESCRIPTION
**Pull request by jessica-mitchell**
This PR adds docs for version 3.5 of NEST
it highlights
* NEST SONATA
* new model: spike_train_injector
* new hpc docs
* run notebooks from pynest examples

### CODE REVIEW

File: doc/htmldoc/whats_new/v3.5/index.rst [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+* :ref:`optimize_performance`
+* :ref:`benchmark`
+
+New model: spike_train_injector
+-------------------------------
+
+The :doc:`spike_train_injector <../../../../models/spike_train_injector>` emits spikes at prescribed spike times which are given as an array.
+
+We recommend its use in multithreaded simulations where spike emitting neurons, in a somewhat large external population, are modeled on an individual-basis.
```
Comment (gtrensch) [inline doc/htmldoc/whats_new/v3.5/index.rst:63]:
```suggestion
We recommend its use in multithreaded simulations where spike-emitting neurons, in a somewhat large external population, are modeled on an individual basis.
```

**Comment by jessica-mitchell** [comment]
@gtrensch @terhorstd see output: https://nest-simulator--2838.org.readthedocs.build/en/2838/whats_new/v3.5/index.html#release-3-5

**Review by gtrensch — APPROVED** [review]
Looks good to me!

**Review by terhorstd — APPROVED** [review]
Nice! Thanks for the nice summary!
:rocket:

**Pull request merged**